File: RepoTransBench/target_projects/Java/C#/rohanpadhye_vasco/tests/original/ContextTest.cs
using Xunit;

namespace Vasco.Tests.Original
{
    public class ContextTest
    {
        [Fact]
        public void TestContextEquals()
        {
            var c1 = new Context("A", 1);
            var c2 = new Context("A", 1);
            var c3 = new Context("B", 2);

            Assert.Equal(c1, c2);
            Assert.NotEqual(c1, c3);
        }

        [Fact]
        public void TestContextHashCode()
        {
            var c1 = new Context("A", 1);
            var c2 = new Context("A", 1);
            Assert.Equal(c1.GetHashCode(), c2.GetHashCode());
        }

        [Fact]
        public void TestNullContext()
        {
            var c1 = new Context(null, 0);
            var c2 = new Context(null, 0);
            Assert.Equal(c1, c2);
        }

        [Fact]
        public void TestContextToString()
        {
            var c1 = new Context("Method", 5);
            Assert.Contains("Method", c1.ToString());
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/rohanpadhye_vasco/tests/original/context_test.go
package original

import (
	"strings"
	"testing"
	"vasco"
)

func TestContextEquals(t *testing.T) {
	c1 := &vasco.Context{Method: "A", ID: 1}
	c2 := &vasco.Context{Method: "A", ID: 1}
	c3 := &vasco.Context{Method: "B", ID: 2}

	if !c1.Equal(c2) {
		t.Error("Context should equal")
	}
	if c1.Equal(c3) {
		t.Error("Context should not equal")
	}
}

func TestContextHashCode(t *testing.T) {
	c1 := &vasco.Context{Method: "A", ID: 1}
	c2 := &vasco.Context{Method: "A", ID: 1}

	if c1.Hash() != c2.Hash() {
		t.Error("Hash codes must match")
	}
}

func TestNullContext(t *testing.T) {
	c1 := &vasco.Context{Method: nil, ID: 0}
	c2 := &vasco.Context{Method: nil, ID: 0}
	if !c1.Equal(c2) {
		t.Error("Contexts with nil Method should equal")
	}
}

func TestContextToString(t *testing.T) {
	c1 := &vasco.Context{Method: "Method", ID: 5}
	if !strings.Contains(c1.String(), "Method") {
		t.Error("String should contain method name")
	}
}